### PR TITLE
[IMPROVE][APPS] Apps list page on servers without internet connection

### PR DESCRIPTION
**Pull request by thassiov**
## Proposed changes

Test the error for connectivity problems and show the message "Could not reach the Marketplace to get Apps info" to better inform the user about the situation. 

## How to test or reproduce

- Take the Rocket.Chat box offline
- Go to Administration -> Apps (the messages appear in the red toast)

## Screenshots
undefined

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Hotfix (a major bugfix that has to be merged asap)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/apps/server/communication/rest.js [hunk 1]
```diff
@@ -60,6 +60,26 @@ export class AppsRestApi {
...
+			if (e.response.statusCode >= 500 && e.response.statusCode <= 599) {
+				return API.v1.internalError();
+			}
+
+			if (e.response.statusCode === 404) {
+				return API.v1.notFound();
+			}
+
+			return API.v1.failure();
+		};
```
Comment (thassiov) [inline app/apps/server/communication/rest.js:81]: Move the function up to avoid "used before declared" issues

**Comment by d-gubert** [comment]
I wonder how useful it is to show this message in an airgapped env.... @RocketChat/backend @RocketChat/frontend do you have an idea of how we could ignore it properly? Assuming we can't simply not show this kind of error because it might be legit in non-airgapped envs

**Comment by sampaiodiego** [comment]
maybe that's an UX issue? if instead of showing a error toast we actually show a yellow banner saying "we couldn't reach official marketplace, listing installed apps only" that can be dismissed it would be good enough?

**Comment by d-gubert** [comment]
> show a yellow banner saying "we couldn't reach official marketplace, listing installed apps only" that can be dismissed

It looks like a good solution in my opinion

**Review by pierre-lehnen-rc — DISMISSED** [review]

**Comment by sampaiodiego** [comment]
> > show a yellow banner saying "we couldn't reach official marketplace, listing installed apps only" that can be dismissed
> 
> It looks like a good solution in my opinion

should we merge that way it is and do what was proposed here as an improvement later on?

**Comment by d-gubert** [comment]
The latest changes do not apply the suggestion?

**Review by sampaiodiego — CHANGES_REQUESTED** [review]
During my tests in my local instance, turning off wi-fi and opening either Apps or Marketplace screens still gave me "cannot read property data of undefined"

**Review by sampaiodiego — APPROVED** [review]

**Pull request merged**